### DISCUSSION TITLE
Call to a member function disk() on null

### DISCUSSION THREAD

**Discussion by AlinaNerlich-dev** [statamic/cms#4541] [Troubleshooting] [answered]
Problem occurs when I try to send my contact form including a file upload...

Call to a member function disk() on null {"userId":"xxx","exception":"[object] (Error(code: 0): Call to a member function disk() on null at /var/www/clients/client1/web195/web/site/vendor/statamic/cms/src/Assets/Asset.php:107)

function around 107 of Asset.php:

/**
     * Get the container's filesystem disk instance.
     *
     * @return \Statamic\Filesystem\FlysystemAdapter
     */
    public function disk()
    {
        return $this->container()->disk(); (this is line 107)
    }

Anyone an idea what to do?

**Comment by duncanmcclean** [comment, marked as answer]
Does the assets field in the form have an asset container selected? (Look in your blueprint)

If not, select one and maybe that'll go away!

**Reply by AlinaNerlich-dev** [reply]
Unfortunately not the solution... It was now working once after doing 'php please cache:clear' but now its the same error

**Reply by AlinaNerlich-dev** [reply]
Ahhhhh - you were absolutely right! I misunderstood - thanks so much!!!!

**Reply by duncanmcclean** [reply]
No worries, glad I could help. Hope you have a great rest of your day!

**Comment by zunnur-tiara** [comment]
Maybe you could try `php artisan optimize:clear`.

**Reply by AlinaNerlich-dev** [reply]
Thanks for the input - I tried but unfortunately same error...
